hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneLoadGenerator.java

```diff
@@ -201,11 +210,12 @@ private void startAgedFilesLoad(long runTimeMillis) {
       String keyName = null;
       try {
         if (agedWriteProbability.isTrue()) {
-          keyName = writeData(agedFileWrittenIndex.incrementAndGet(),
+          keyName = writeData(agedFileWrittenIndex.getAndIncrement(),
```

Comment:
Aged file index always start with 0, so we will never read the first file.

Reply:
Can you please clarify what's the issue here?

Random index selection gives a number in `[0, currentIndex)` (see `RandomUtils.nextInt(0, currentIndex)`).

Previously files were written starting at 1 and `currentIndex` equaled the last written key index.  Now `currentIndex` is last written key index + 1, so the random choice always produces an existing key index.

Reply:
With getAndIncrement, the first key is written with index 0.
in randomKeyToRead, for current index of 0, we return the Optional.empty(). And hence the first key will never be read.

Reply:
Before key with index=0 is written, the current index is incremented to 1.  Please see my [example log](https://github.com/apache/hadoop/pull/1287#issue-306801835) for reads of the first key right after the first write (and even after the second one).

Reply:
got that, Thanks for the explanation. This comment can be ignored.